leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/LeshanClient.java

```diff
@@ -98,18 +103,33 @@ public LeshanClient(final String endpoint, final InetSocketAddress localAddress,
         // Create sender
         requestSender = new CaliforniumLwM2mClientRequestSender(secureEndpoint, nonSecureEndpoint);
 
-        // Create registration engine
-        bootstrapHandler = new BootstrapHandler(this.objectEnablers);
-        bootstrapHandler.addBootstrapListener(new BootstrapListener() {
```

Comment:
The BootstrapListener is probably not used anymore. We should remove it and rely on the LwM2mClientObserver events.